src/pages/settings/Profile/ProfilePage.js

```diff
@@ -80,23 +81,37 @@ class ProfilePage extends Component {
             selectedTimezone: lodashGet(props.myPersonalDetails.timezone, 'selected', CONST.DEFAULT_TIME_ZONE.selected),
             isAutomaticTimezone: lodashGet(props.myPersonalDetails.timezone, 'automatic', CONST.DEFAULT_TIME_ZONE.automatic),
             logins: this.getLogins(props.user.loginList),
+            avatarImage: null,
+            avatarPreviewURL: lodashGet(this.props.myPersonalDetails, 'avatar', OptionsListUtils.getDefaultAvatar(this.props.myPersonalDetails.login)),
+            isAvatarUpdated: false,
         };
+
         this.getLogins = this.getLogins.bind(this);
         this.setAutomaticTimezone = this.setAutomaticTimezone.bind(this);
         this.updatePersonalDetails = this.updatePersonalDetails.bind(this);
         this.validateInputs = this.validateInputs.bind(this);
+        this.setAvatar = this.setAvatar.bind(this);
+        this.removeAvatar = this.removeAvatar.bind(this);
     }
 
     componentDidUpdate(prevProps) {
+        let stateToUpdate = {};
+
         // Recalculate logins if loginList has changed
-        if (this.props.user.loginList === prevProps.user.loginList) {
+        if (this.props.user.loginList !== prevProps.user.loginList) {
```

Comment:
Can this ever be false? I'm pretty sure in JS you can never check if arrays (objects) are equal with `===` or `!==` because they're pointers